### PR TITLE
Use matrix in Github Actions

### PR DESCRIPTION
**Pull request by mienaikage**
Add macOS and use a matrix for running jobs on each OS

### CODE REVIEW

**Comment by ekingery** [comment]
This looks fine to me, but ideally we could get someone from @exercism/github-actions to review. I'll approve after a couple days if it doesn't get picked up.

**Comment by SleeplessByte** [comment]
Sascha or Erik (or someone from the team)  will likely pick this up tomorrow at the Hackathon!

File: .github/workflows/exercises.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        if: runner.os != 'Linux'
+        run: prove6 --jobs 2 (Get-Item -Path exercises/*/.meta/solutions/)
+        shell: pwsh
+
+      - name: Rakudo Star tests with docker run
+        if: runner.os == 'Linux'
+        run: docker run
+             --volume $PWD:/exercism
+             rakudo-star /bin/bash -c 'prove6 --jobs 2
+               /exercism/exercises/*/.meta/solutions/'
```
Comment (cmccandless) [inline .github/workflows/exercises.yml:43]: Steps can be specified to run containers. This should work as well or better.
```suggestion
        uses: docker://rakudo-star:latest
        with:
          args: /bin/bash -c 'prove6 --jobs 2 /github/workspace/exercises/*/.meta/solutions/'
```

**Review by cmccandless — CHANGES_REQUESTED** [review]
I took a look at the [builds][sample-builds] in your fork. Yeesh, that `choco install` takes a long time (compared to the install step on macOS and the docker pull on Ubuntu). Unfortunately, I don't have any suggestions to speed it up.

[sample-builds]: https://github.com/mienaikage/exercism-raku/runs/1334321972?check_suite_focus=true

**Comment by cmccandless** [comment]
Was there a reason you didn't use the `apt-get` package for the Ubuntu build?

File: .github/workflows/exercises.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+      # https://github.com/Leont/getopt-long6/issues/10
+      - name: Install Getopt::Long:ver<0.1.7>
+        if: runner.os != 'Linux'
+        run: zef install https://github.com/Leont/getopt-long6.git@0.1.7
+
+      - name: Rakudo Star tests with prove6
+        if: runner.os != 'Linux'
+        run: prove6 --jobs 2 (Get-Item -Path exercises/*/.meta/solutions/)
+        shell: pwsh
```
Comment (ErikSchierboom) [inline .github/workflows/exercises.yml:36]: Interesting. The `pwsh` shell is also available on macOS? Nice.

**Review by ErikSchierboom — APPROVED** [review]
LGTM

File: .github/workflows/exercises.yml (hunk 2 above)
Comment (cmccandless) [reply .github/workflows/exercises.yml:36]: Bash and pwsh are the only shells supported on all three OS.

**Review by cmccandless — APPROVED** [review]
LGTM

**Pull request merged**